### ISSUE TITLE
WandbLogger causes the program to crash without an error

### ISSUE THREAD

**Issue by sudomaze** [PyTorchLightning/pytorch-lightning#12463]
## 🐛 Bug

In my simple setup, I have been using TensorboardLogger and other loggers and everything worked perfectly. However, when I tried to use WandbLogger, the program crashes without any error. It just prints  a dictionary and stops.

### To Reproduce

```python
{'_identity': (1,), '_config': {'authkey': b'i\x01FT\x8b\xe0\xf6e}\xe7\xce\xe7\xa1\xb5\x9a\x9bF2A?\x95\x95\xe1\x85I\x82a\xa4\xef4\xc3=', 'semprefix': '/mp'}, '_parent_pid': 4007964, '_parent_name': 'MainProcess', '_popen': None, '_closed': False, '_target': <function wandb_internal at 0x7f4792fd4550>, '_args': (), '_kwargs': {'settings': {'_args': [], '_cli_only_mode': None, '_colab': False, '_config_dict': None, '_console': <SettingsConsole.REDIRECT: 2>, '_cuda': None, '_disable_meta': None, '_disable_stats': None, '_disable_viewer': None, '_except_exit': None, '_executable': '/home/mazen/miniconda3/envs/jdt/bin/python', '_internal_check_process': 8, '_internal_queue_timeout': 2, '_jupyter': False, '_jupyter_name': None, '_jupyter_path': None, '_jupyter_root': None, '_kaggle': False, '_noop': False, '_offline': False, '_os': 'Linux-5.13.0-28-generic-x86_64-with-glibc2.31', '_platform': 'linux', '_python': '3.9.7', '_require_service': None, '_runqueue_item_id': None, '_save_requirements': True, '_service_transport': None, '_start_datetime': datetime.datetime(2022, 3, 25, 19, 17, 2, 616686), '_start_time': 1648261022.616686, '_tmp_code_dir': '/home/mazen/Projects/pl_jdt/output/wandb/run-20220325_191702-1wdniopy/tmp/code', '_tracelog': None, '_unsaved_keys': None, '_windows': False, 'allow_val_change': None, 'anonymous': None, 'api_key': None, 'base_url': 'https://api.wandb.ai', 'code_dir': None, 'config_paths': None, 'console': 'auto', 'deployment': 'cloud', 'disable_code': None, 'disable_git': False, 'disabled': False, 'docker': None, 'email': 'mazen.ota@gmail.com', 'entity': None, 'files_dir': '/home/mazen/Projects/pl_jdt/output/wandb/run-20220325_191702-1wdniopy/files', 'force': None, 'git_remote': 'origin', 'heartbeat_seconds': 30, 'host': 'mazen-HP-Z640-Workstation', 'ignore_globs': (), 'is_local': False, 'label_disable': None, 'launch': None, 'launch_config_path': None, 'log_dir': '/home/mazen/Projects/pl_jdt/output/wandb/run-20220325_191702-1wdniopy/logs', 'log_internal': '/home/mazen/Projects/pl_jdt/output/wandb/run-20220325_191702-1wdniopy/logs/debug-internal.log', 'log_symlink_internal': '/home/mazen/Projects/pl_jdt/output/wandb/debug-internal.log', 'log_symlink_user': '/home/mazen/Projects/pl_jdt/output/wandb/debug.log', 'log_user': '/home/mazen/Projects/pl_jdt/output/wandb/run-20220325_191702-1wdniopy/logs/debug.log', 'login_timeout': None, 'magic': None, 'mode': 'online', 'notebook_name': None, 'problem': 'fatal', 'program': '/home/mazen/Projects/pl_jdt/scripts/train.py', 'program_relpath': 'scripts/train.py', 'project': 'mnist_training_test', 'project_url': '', 'quiet': None, 'reinit': None, 'relogin': None, 'resume': 'allow', 'resume_fname': '/home/mazen/Projects/pl_jdt/output/wandb/wandb-resume.json', 'resumed': False, 'root_dir': '/home/mazen/Projects/pl_jdt/output', 'run_group': None, 'run_id': '1wdniopy', 'run_job_type': None, 'run_mode': 'run', 'run_name': None, 'run_notes': None, 'run_tags': None, 'run_url': '', 'sagemaker_disable': None, 'save_code': True, 'settings_system': '/home/mazen/.config/wandb/settings', 'settings_workspace': '/home/mazen/Projects/pl_jdt/output/wandb/settings', 'show_colors': None, 'show_emoji': None, 'show_errors': True, 'show_info': True, 'show_warnings': True, 'silent': False, 'start_method': None, 'strict': None, 'summary_errors': None, 'summary_warnings': 5, 'sweep_id': None, 'sweep_param_path': None, 'sweep_url': '', 'symlink': True, 'sync_dir': '/home/mazen/Projects/pl_jdt/output/wandb/run-20220325_191702-1wdniopy', 'sync_file': '/home/mazen/Projects/pl_jdt/output/wandb/run-20220325_191702-1wdniopy/run-1wdniopy.wandb', 'sync_symlink_latest': '/home/mazen/Projects/pl_jdt/output/wandb/latest-run', 'system_sample': 15, 'system_sample_seconds': 2, 'timespec': '20220325_191702', 'tmp_dir': '/home/mazen/Projects/pl_jdt/output/wandb/run-20220325_191702-1wdniopy/tmp', 'username': 'mazen', 'wandb_dir': '/home/mazen/Projects/pl_jdt/output/wandb/', '_log_level': 10}, 'record_q': <multiprocessing.queues.Queue object at 0x7f478fceb430>, 'result_q': <multiprocessing.queues.Queue object at 0x7f478fcfb100>, 'user_pid': 4007964}, '_name': 'wandb_internal'} <_io.BytesIO object at 0x7f478f477e50>
```
### Expected behavior

I am expecting for WandbLogger to work as the other loggers work as expected.

### Environment
```
* CUDA:
        - GPU:
                - NVIDIA TITAN X (Pascal)
                - NVIDIA TITAN X (Pascal)
        - available:         True
        - version:           11.3
* Packages:
        - numpy:             1.21.2
        - pyTorch_debug:     False
        - pyTorch_version:   1.11.0
        - pytorch-lightning: 1.5.10
        - tqdm:              4.62.3
* System:
        - OS:                Linux
        - architecture:
                - 64bit
                - ELF
        - processor:         x86_64
        - python:            3.9.7
        - version:           #31~20.04.1-Ubuntu SMP Wed Jan 19 14:08:10 UTC 2022
```
- PyTorch Lightning Version (e.g., 1.5.0): 1.5.10
- PyTorch Version (e.g., 1.10): 1.11.0
- Python version (e.g., 3.9): 3.9.7
- OS (e.g., Linux): Linux (#31~20.04.1-Ubuntu SMP Wed Jan 19 14:08:10 UTC 2022)
- CUDA/cuDNN version: 11.3
- GPU models and configuration: NVIDIA TITAN X (Pascal)
- How you installed PyTorch (`conda`, `pip`, source): conda
- Any other relevant information: My script works on a single and DDP. I have been trying to have WandbLogger to work on both but I am getting the same error.

### Additional context

I am refactoring my code so I have updated PyTorch to 1.11 and PyTorch Lightning to 1.5.10. I have started developing setup by step so I can ensure that everything is working as expected. I am using `Hydra` and `Rich` as the only external libraries (besides PyTorch, PyTorch Lightning, torchvision, and torchmetric).

cc @awaelchli @morganmcg1 @AyushExel @borisdayma @scottire @manangoel99

**Comment by morganmcg1** [comment]
Hey, I work at Weights & Biases, does the simple example at http://wandb.me/lightning work for you?

To help diagnose this it might also be helpful to only call wandb on the main process (e.g. `gpu_0`) as there might be something related to your distributed setup going on.

If you can share any other code about how you're logging with wandb that would be helpful to see too :)

**Comment by sudomaze** [comment]
Hi, I just copied and pasted the same exact code that you have shared with me (the notebook) and it resulted to the same error.

<details>
<summary>Code</summary>

```python
from pytorch_lightning.loggers import WandbLogger
from pytorch_lightning import Trainer

wandb_logger = WandbLogger()
trainer = Trainer(logger=wandb_logger)

import wandb
wandb.login()

from torchvision.datasets import MNIST
from torchvision import transforms
from torch.utils.data import DataLoader, random_split

transform = transforms.Compose([
            transforms.ToTensor(),
            transforms.Normalize((0.1307,), (0.3081,))])

dataset = MNIST(root="./MNIST", download=True, transform=transform)
training_set, validation_set = random_split(dataset, [55000, 5000])

training_loader = DataLoader(training_set, batch_size=64, shuffle=True)
validation_loader = DataLoader(validation_set, batch_size=64)

import torch
from torch.nn import Linear, CrossEntropyLoss, functional as F
from torch.optim import Adam
from torchmetrics.functional import accuracy
from pytorch_lightning import LightningModule

class MNIST_LitModule(LightningModule):

    def __init__(self, n_classes=10, n_layer_1=128, n_layer_2=256, lr=1e-3):
        '''method used to define our model parameters'''
        super().__init__()

        # mnist images are (1, 28, 28) (channels, width, height)
        self.layer_1 = Linear(28 * 28, n_layer_1)
        self.layer_2 = Linear(n_layer_1, n_layer_2)
        self.layer_3 = Linear(n_layer_2, n_classes)

        # loss
        self.loss = CrossEntropyLoss()

        # optimizer parameters
        self.lr = lr

        # save hyper-parameters to self.hparams (auto-logged by W&B)
        self.save_hyperparameters()

    def forward(self, x):
        '''method used for inference input -> output'''

        batch_size, channels, width, height = x.size()

        # (b, 1, 28, 28) -> (b, 1*28*28)
        x = x.view(batch_size, -1)

        # let's do 3 x (linear + relu)
        x = self.layer_1(x)
        x = F.relu(x)
        x = self.layer_2(x)
        x = F.relu(x)
        x = self.layer_3(x)

        return x

    def training_step(self, batch, batch_idx):
        '''needs to return a loss from a single batch'''
        _, loss, acc = self._get_preds_loss_accuracy(batch)

        # Log loss and metric
        self.log('train_loss', loss)
        self.log('train_accuracy', acc)

        return loss

    def validation_step(self, batch, batch_idx):
        '''used for logging metrics'''
        preds, loss, acc = self._get_preds_loss_accuracy(batch)

        # Log loss and metric
        self.log('val_loss', loss)
        self.log('val_accuracy', acc)

        # Let's return preds to use it in a custom callback
        return preds

    def test_step(self, batch, batch_idx):
        '''used for logging metrics'''
        _, loss, acc = self._get_preds_loss_accuracy(batch)

        # Log loss and metric
        self.log('test_loss', loss)
        self.log('test_accuracy', acc)
    
    def configure_optimizers(self):
        '''defines model optimizer'''
        return Adam(self.parameters(), lr=self.lr)
    
    def _get_preds_loss_accuracy(self, batch):
        '''convenience function since train/valid/test steps are similar'''
        x, y = batch
        logits = self(x)
        preds = torch.argmax(logits, dim=1)
        loss = self.loss(logits, y)
        acc = accuracy(preds, y)
        return preds, loss, acc

model = MNIST_LitModule(n_layer_1=128, n_layer_2=128)

from pytorch_lightning.callbacks import ModelCheckpoint

checkpoint_callback = ModelCheckpoint(monitor='val_accuracy', mode='max')

from pytorch_lightning.loggers import WandbLogger
from pytorch_lightning import Trainer

wandb_logger = WandbLogger(project='MNIST', # group runs in "MNIST" project
                           log_model='all') # log all new checkpoints during training

from pytorch_lightning.callbacks import Callback
 
class LogPredictionsCallback(Callback):
    
    def on_validation_batch_end(
        self, trainer, pl_module, outputs, batch, batch_idx, dataloader_idx):
        """Called when the validation batch ends."""
 
        # `outputs` comes from `LightningModule.validation_step`
        # which corresponds to our model predictions in this case
        
        # Let's log 20 sample image predictions from first batch
        if batch_idx == 0:
            n = 20
            x, y = batch
            images = [img for img in x[:n]]
            captions = [f'Ground Truth: {y_i} - Prediction: {y_pred}' for y_i, y_pred in zip(y[:n], outputs[:n])]
            
            # Option 1: log images with `WandbLogger.log_image`
            wandb_logger.log_image(key='sample_images', images=images, caption=captions)

            # Option 2: log predictions as a Table
            columns = ['image', 'ground truth', 'prediction']
            data = [[wandb.Image(x_i), y_i, y_pred] for x_i, y_i, y_pred in list(zip(x[:n], y[:n], outputs[:n]))]
            wandb_logger.log_table(key='sample_table', columns=columns, data=data)

log_predictions_callback = LogPredictionsCallback()

trainer = Trainer(
    logger=wandb_logger,                    # W&B integration
    callbacks=[log_predictions_callback,    # logging of sample predictions
               checkpoint_callback],        # our model checkpoint callback
    max_epochs=5)                           # number of epochs

trainer.fit(model, training_loader, validation_loader)

wandb.finish()
```
</details>

<details>

<summary>Output</summary>

```shell
python main.py 
GPU available: True, used: False
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
/home/mazen/miniconda3/envs/jdt/lib/python3.9/site-packages/pytorch_lightning/trainer/trainer.py:1584: UserWarning: GPU available but not used. Set the gpus flag in your trainer `Trainer(gpus=1)` or script `--gpus=1`.
  rank_zero_warn(
wandb: Currently logged in as: sudomaze (use `wandb login --relogin` to force relogin)
Downloading http://yann.lecun.com/exdb/mnist/train-images-idx3-ubyte.gz
Downloading http://yann.lecun.com/exdb/mnist/train-images-idx3-ubyte.gz to ./MNIST/MNIST/raw/train-images-idx3-ubyte.gz
9913344it [00:00, 34873741.51it/s]                                                            
Extracting ./MNIST/MNIST/raw/train-images-idx3-ubyte.gz to ./MNIST/MNIST/raw

Downloading http://yann.lecun.com/exdb/mnist/train-labels-idx1-ubyte.gz
Downloading http://yann.lecun.com/exdb/mnist/train-labels-idx1-ubyte.gz to ./MNIST/MNIST/raw/train-labels-idx1-ubyte.gz
29696it [00:00, 41052752.66it/s]                                                              
Extracting ./MNIST/MNIST/raw/train-labels-idx1-ubyte.gz to ./MNIST/MNIST/raw

Downloading http://yann.lecun.com/exdb/mnist/t10k-images-idx3-ubyte.gz
Downloading http://yann.lecun.com/exdb/mnist/t10k-images-idx3-ubyte.gz to ./MNIST/MNIST/raw/t10k-images-idx3-ubyte.gz
1649664it [00:00, 13475043.55it/s]                                                            
Extracting ./MNIST/MNIST/raw/t10k-images-idx3-ubyte.gz to ./MNIST/MNIST/raw

Downloading http://yann.lecun.com/exdb/mnist/t10k-labels-idx1-ubyte.gz
Downloading http://yann.lecun.com/exdb/mnist/t10k-labels-idx1-ubyte.gz to ./MNIST/MNIST/raw/t10k-labels-idx1-ubyte.gz
5120it [00:00, 26317201.57it/s]                                                               
Extracting ./MNIST/MNIST/raw/t10k-labels-idx1-ubyte.gz to ./MNIST/MNIST/raw

GPU available: True, used: False
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
/home/mazen/miniconda3/envs/jdt/lib/python3.9/site-packages/pytorch_lightning/trainer/trainer.py:1584: UserWarning: GPU available but not used. Set the gpus flag in your trainer `Trainer(gpus=1)` or script `--gpus=1`.
  rank_zero_warn(
{'_identity': (1,), '_config': {'authkey': b'\x16(\xec\xc7\xf2\xadc&\x9b\xca{\xe3\xc5\x9c5\x06\xe6h\xb8\xabO\xd0\x82\x02uh\x9d\x93\x92\x15\x05\x80', 'semprefix': '/mp'}, '_parent_pid': 41540, '_parent_name': 'MainProcess', '_popen': None, '_closed': False, '_target': <function wandb_internal at 0x7fb20a907ee0>, '_args': (), '_kwargs': {'settings': {'_args': [], '_cli_only_mode': None, '_colab': False, '_config_dict': None, '_console': <SettingsConsole.REDIRECT: 2>, '_cuda': None, '_disable_meta': None, '_disable_stats': None, '_disable_viewer': None, '_except_exit': None, '_executable': '/home/mazen/miniconda3/envs/jdt/bin/python', '_internal_check_process': 8, '_internal_queue_timeout': 2, '_jupyter': False, '_jupyter_name': None, '_jupyter_path': None, '_jupyter_root': None, '_kaggle': False, '_noop': False, '_offline': False, '_os': 'Linux-5.13.0-28-generic-x86_64-with-glibc2.31', '_platform': 'linux', '_python': '3.9.7', '_require_service': None, '_runqueue_item_id': None, '_save_requirements': True, '_service_transport': None, '_start_datetime': datetime.datetime(2022, 3, 25, 22, 59, 15, 160887), '_start_time': 1648274355.1608872, '_tmp_code_dir': '/home/mazen/Projects/pl_jdt/tmp/wandb/run-20220325_225915-3ulzi5qr/tmp/code', '_tracelog': None, '_unsaved_keys': None, '_windows': False, 'allow_val_change': None, 'anonymous': None, 'api_key': None, 'base_url': 'https://api.wandb.ai', 'code_dir': None, 'config_paths': None, 'console': 'auto', 'deployment': 'cloud', 'disable_code': None, 'disable_git': False, 'disabled': False, 'docker': None, 'email': 'mazen.ota@gmail.com', 'entity': None, 'files_dir': '/home/mazen/Projects/pl_jdt/tmp/wandb/run-20220325_225915-3ulzi5qr/files', 'force': None, 'git_remote': 'origin', 'heartbeat_seconds': 30, 'host': 'mazen-HP-Z640-Workstation', 'ignore_globs': (), 'is_local': False, 'label_disable': None, 'launch': None, 'launch_config_path': None, 'log_dir': '/home/mazen/Projects/pl_jdt/tmp/wandb/run-20220325_225915-3ulzi5qr/logs', 'log_internal': '/home/mazen/Projects/pl_jdt/tmp/wandb/run-20220325_225915-3ulzi5qr/logs/debug-internal.log', 'log_symlink_internal': '/home/mazen/Projects/pl_jdt/tmp/wandb/debug-internal.log', 'log_symlink_user': '/home/mazen/Projects/pl_jdt/tmp/wandb/debug.log', 'log_user': '/home/mazen/Projects/pl_jdt/tmp/wandb/run-20220325_225915-3ulzi5qr/logs/debug.log', 'login_timeout': None, 'magic': None, 'mode': 'online', 'notebook_name': None, 'problem': 'fatal', 'program': '/home/mazen/Projects/pl_jdt/tmp/main.py', 'program_relpath': 'main.py', 'project': 'MNIST', 'project_url': '', 'quiet': None, 'reinit': None, 'relogin': None, 'resume': 'allow', 'resume_fname': '/home/mazen/Projects/pl_jdt/tmp/wandb/wandb-resume.json', 'resumed': False, 'root_dir': '/home/mazen/Projects/pl_jdt/tmp', 'run_group': None, 'run_id': '3ulzi5qr', 'run_job_type': None, 'run_mode': 'run', 'run_name': None, 'run_notes': None, 'run_tags': None, 'run_url': '', 'sagemaker_disable': None, 'save_code': True, 'settings_system': '/home/mazen/.config/wandb/settings', 'settings_workspace': '/home/mazen/Projects/pl_jdt/tmp/wandb/settings', 'show_colors': None, 'show_emoji': None, 'show_errors': True, 'show_info': True, 'show_warnings': True, 'silent': False, 'start_method': None, 'strict': None, 'summary_errors': None, 'summary_warnings': 5, 'sweep_id': None, 'sweep_param_path': None, 'sweep_url': '', 'symlink': True, 'sync_dir': '/home/mazen/Projects/pl_jdt/tmp/wandb/run-20220325_225915-3ulzi5qr', 'sync_file': '/home/mazen/Projects/pl_jdt/tmp/wandb/run-20220325_225915-3ulzi5qr/run-3ulzi5qr.wandb', 'sync_symlink_latest': '/home/mazen/Projects/pl_jdt/tmp/wandb/latest-run', 'system_sample': 15, 'system_sample_seconds': 2, 'timespec': '20220325_225915', 'tmp_dir': '/home/mazen/Projects/pl_jdt/tmp/wandb/run-20220325_225915-3ulzi5qr/tmp', 'username': 'mazen', 'wandb_dir': '/home/mazen/Projects/pl_jdt/tmp/wandb/', '_log_level': 10}, 'record_q': <multiprocessing.queues.Queue object at 0x7fb2002f1130>, 'result_q': <multiprocessing.queues.Queue object at 0x7fb2002f1760>, 'user_pid': 41540}, '_name': 'wandb_internal'} <_io.BytesIO object at 0x7fb2002fbef0>
```
</details>

**Comment by manangoel99** [comment]
Hi @sudomaze . Could you also please share the version of wandb that you're using

**Comment by sudomaze** [comment]
```wandb=0.12.11                  pypi_0    pypi```

**Comment by manangoel99** [comment]
I tried recreating the error on my end with the same version of the libraries but it is working for me. Could you try running the following python script

```python
import wandb

wandb.init()
wandb.log({"x": 1})
wandb.finish()
```

**Comment by sudomaze** [comment]
Same error.
```shell
python w.py 
wandb: Currently logged in as: sudomaze (use `wandb login --relogin` to force relogin)
{'_identity': (1,), '_config': {'authkey': b's\xde\x0fT\\\xf741X\xce\xfa\xd90\xe4b\xf2\xf0x\x1cHUJ\xf8\x06\x882\xf9\xed\x08\xfa\xf0\x93', 'semprefix': '/mp'}, '_parent_pid': 109223, '_parent_name': 'MainProcess', '_popen': None, '_closed': False, '_target': <function wandb_internal at 0x7fab7a3fcb80>, '_args': (), '_kwargs': {'settings': {'_args': [], '_cli_only_mode': None, '_colab': False, '_config_dict': None, '_console': <SettingsConsole.REDIRECT: 2>, '_cuda': None, '_disable_meta': None, '_disable_stats': None, '_disable_viewer': None, '_except_exit': None, '_executable': '/home/mazen/miniconda3/envs/jdt/bin/python', '_internal_check_process': 8, '_internal_queue_timeout': 2, '_jupyter': False, '_jupyter_name': None, '_jupyter_path': None, '_jupyter_root': None, '_kaggle': False, '_noop': False, '_offline': False, '_os': 'Linux-5.13.0-28-generic-x86_64-with-glibc2.31', '_platform': 'linux', '_python': '3.9.7', '_require_service': None, '_runqueue_item_id': None, '_save_requirements': True, '_service_transport': None, '_start_datetime': datetime.datetime(2022, 3, 26, 0, 6, 46, 493490), '_start_time': 1648278406.4934897, '_tmp_code_dir': '/home/mazen/Projects/pl_jdt/tmp/wandb/run-20220326_000646-2xe8l7b6/tmp/code', '_tracelog': None, '_unsaved_keys': None, '_windows': False, 'allow_val_change': None, 'anonymous': None, 'api_key': None, 'base_url': 'https://api.wandb.ai', 'code_dir': None, 'config_paths': None, 'console': 'auto', 'deployment': 'cloud', 'disable_code': None, 'disable_git': False, 'disabled': False, 'docker': None, 'email': 'mazen.ota@gmail.com', 'entity': None, 'files_dir': '/home/mazen/Projects/pl_jdt/tmp/wandb/run-20220326_000646-2xe8l7b6/files', 'force': None, 'git_remote': 'origin', 'heartbeat_seconds': 30, 'host': 'mazen-HP-Z640-Workstation', 'ignore_globs': (), 'is_local': False, 'label_disable': None, 'launch': None, 'launch_config_path': None, 'log_dir': '/home/mazen/Projects/pl_jdt/tmp/wandb/run-20220326_000646-2xe8l7b6/logs', 'log_internal': '/home/mazen/Projects/pl_jdt/tmp/wandb/run-20220326_000646-2xe8l7b6/logs/debug-internal.log', 'log_symlink_internal': '/home/mazen/Projects/pl_jdt/tmp/wandb/debug-internal.log', 'log_symlink_user': '/home/mazen/Projects/pl_jdt/tmp/wandb/debug.log', 'log_user': '/home/mazen/Projects/pl_jdt/tmp/wandb/run-20220326_000646-2xe8l7b6/logs/debug.log', 'login_timeout': None, 'magic': None, 'mode': 'online', 'notebook_name': None, 'problem': 'fatal', 'program': '/home/mazen/Projects/pl_jdt/tmp/w.py', 'program_relpath': 'w.py', 'project': None, 'project_url': '', 'quiet': None, 'reinit': None, 'relogin': None, 'resume': None, 'resume_fname': '/home/mazen/Projects/pl_jdt/tmp/wandb/wandb-resume.json', 'resumed': False, 'root_dir': '/home/mazen/Projects/pl_jdt/tmp', 'run_group': None, 'run_id': '2xe8l7b6', 'run_job_type': None, 'run_mode': 'run', 'run_name': None, 'run_notes': None, 'run_tags': None, 'run_url': '', 'sagemaker_disable': None, 'save_code': True, 'settings_system': '/home/mazen/.config/wandb/settings', 'settings_workspace': '/home/mazen/Projects/pl_jdt/tmp/wandb/settings', 'show_colors': None, 'show_emoji': None, 'show_errors': True, 'show_info': True, 'show_warnings': True, 'silent': False, 'start_method': None, 'strict': None, 'summary_errors': None, 'summary_warnings': 5, 'sweep_id': None, 'sweep_param_path': None, 'sweep_url': '', 'symlink': True, 'sync_dir': '/home/mazen/Projects/pl_jdt/tmp/wandb/run-20220326_000646-2xe8l7b6', 'sync_file': '/home/mazen/Projects/pl_jdt/tmp/wandb/run-20220326_000646-2xe8l7b6/run-2xe8l7b6.wandb', 'sync_symlink_latest': '/home/mazen/Projects/pl_jdt/tmp/wandb/latest-run', 'system_sample': 15, 'system_sample_seconds': 2, 'timespec': '20220326_000646', 'tmp_dir': '/home/mazen/Projects/pl_jdt/tmp/wandb/run-20220326_000646-2xe8l7b6/tmp', 'username': 'mazen', 'wandb_dir': '/home/mazen/Projects/pl_jdt/tmp/wandb/', '_log_level': 10}, 'record_q': <multiprocessing.queues.Queue object at 0x7fab7930d4f0>, 'result_q': <multiprocessing.queues.Queue object at 0x7fab78a25970>, 'user_pid': 109223}, '_name': 'wandb_internal'} <_io.BytesIO object at 0x7fab78a2fb80>
```

**Comment by manangoel99** [comment]
In that case, it seems like it's an issue with with your wandb setup. Please run `wandb login --relogin` on the terminal and then run the same python file. It seems like an issue with your wandb setup.

**Comment by sudomaze** [comment]
Same issue. I have removed the pip version and installed the conda version, but the same issue. 

I have found out the problem. It seems that while debugging the DDP approach (as I was doing it manually). I have left the following:
```python
print(process_obj.__dict__, fp); exit()
```
in `python3.9/multiprocessing/popen_spawn_posic.py` line 46.

This isn't a bug in `wandb` nor in `PyTorch Lightning` but I have processed the bug while debugging DDP to build my custom version. It is interesting how other loggers nor the PL's DDP have encountered this issue.

Now everything is working! :D

I do apologize for the confusion. Thanks for your help @morganmcg1 @manangoel99 @akihironitta (I will close the issue).
